types/pg/index.d.ts

```diff
@@ -326,3 +326,5 @@ import * as Pg from ".";
 export const native: typeof Pg | null;
 
 export { DatabaseError } from "pg-protocol";
+
+export type Result = new(rowMode: string, t: typeof types) => QueryResult;
```

Comment:
I think this should be a class, since it is a class in the original library: https://github.com/brianc/node-postgres/blob/master/packages/pg/lib/result.js#L10

Reply:
right, was trying to keep it simple, because getting the class right would be more involved, and for the purpose of this PR, the constructor is enough. I can look into adding the whole class there as well, but it could also be a separate PR? It's not directly related to the feature of this PR...

Reply:
fixed!